### ISSUE TITLE
Logged config does not contain default assigned values

### ISSUE THREAD

**Issue by samiransen23** [Cascella-Group-UiO/HyMD#166]
The config is logged right after parsing. So a value not appearing in `config.tomli` defaults to None/0. This gets logged at the start of `log.txt` which is what users would check to know the configuration they used at a later time. But the value assigned later to this variable (a non-zero value by default that the code uses) does not appear in the logs. This data goes unrecorded from the log file.

The most probable solution, whoever wants to do it:
Call `Logger.rank0.log(logging.INFO, str(config))` after all default values have been assigned instead of calling it right after parsing (which is what the code does now: look in `configure_runtime.py`

**Comment by samiransen23** [comment]
Tested. It works.
Most default values are assigned after the `hamiltonian` call. So it's okay to call `Logger` after that.
But I'm not sure of the following:
1) @Lun4m : dihedral stuff still look blank, unless they're supposed to be so --- my system doesn't have any dihedrals? (See below: `dihedrals:`)
2) @lasse-steinnes : a bunch of stuff are still None --- my system doesn't have any charge. (See below: `coulombtype`, `convergence_type`, etc)

If the values shown below as they should, we need a change to not print a list of `None` and `[ ]` because it's uselessly ugly.

```bash
	Config: /Users/samiransen23/hymd-module-fork/examples/dppc/configtest.tomli
	--------------------------------------------------
	n_steps: 2
	time_step: 0.03
	mesh_size: [50, 50, 50]
	sigma: 0.5
	kappa: 0.05
	box_size: [13, 13, 14]
	n_print: 1
	tau: 0.1
	start_temperature: 323
	target_temperature: 323
	mass: 72.0
	hamiltonian: DefaultWithChi
	domain_decomposition: 5000
	integrator: respa
	respa_inner: 10
	file_name: /Users/samiransen23/hymd-module-fork/examples/dppc/configtest.tomli
	name: DPPC bilayer with ML interaction parameters
	tags: ['bilayer', 'solvent', 'DPPC']
	n_particles: 20336
	max_molecule_size: 15
	n_flush: 1
	thermostat_work: 0.0
	initial_energy: None
	cancel_com_momentum: True
	coulombtype: None
	convergence_type: None
	pol_mixing: None
	dielectric_const: None
	conv_crit: None
	dielectric_type: []
	squaregradient: False
	rho0: None
	a: 9.21
	pressure: True
	barostat: semiisotropic
	tau_p: 1
	alpha_0: None
	n_b: None
	m: [1.0, 1.0, 1.0, 1.0, 1.0]
	command_line_full: /Users/samiransen23/hymd-module-fork/hymd/__main__.py examples/dppc/configtest.tomli examples/dppc/dppc.h5 --verbose --logfile log.txt
	target_pressure:
		P_L: 1
		P_N: 1
	bonds:
		N P: 0.47, 1250.0
		P G: 0.47, 1250.0
		G G: 0.37, 1250.0
		G C: 0.47, 1250.0
		C C: 0.47, 1250.0
	angle_bonds:
		P G G: 120.0, 25.0
		P G C: 180.0, 25.0
		G C C: 180.0, 25.0
		C C C: 180.0, 25.0
	dihedrals:
	chi:
		C W: 42.24
		C G: 10.47
		N W: -3.77
		G W: 4.53
		N P: -9.34
		G P: 8.04
		G N: 1.97
		C P: 14.72
		P W: -1.51
		C N: 13.56
		N N: 0.0
		P P: 0.0
		G G: 0.0
		C C: 0.0
		W W: 0.0
	K_coupl:
```

**Comment by mortele** [comment]
> If the values shown below as they should, we need a change to not print a list of `None` and `[ ]` because it's uselessly ugly.
@samiransen23 Ugly, but not useless. It is good to know exactly what is set and which variables are not specified (left to default `None` or empty lists).

**Comment by samiransen23** [comment]
> > If the values shown below as they should, we need a change to not print a list of `None` and `[ ]` because it's uselessly ugly.
> 
> @samiransen23 Ugly, but not useless. It is good to know exactly what is set and which variables are not specified (left to default `None` or empty lists).

Sure, but also important to know what values are in fact set legibly. Prompts by warnings are hard to hunt down, not even sure if all the prompts mention the default values at all.

**Comment by Lun4m** [comment]
I agree with @mortele here, however does this clutter the terminal output or it's only present in the log file?
Also it makes sense the `dihedrals` section is empty since you don't have any, I don't know what you'd expect?

And could we change `tau -> tau_t` (since now you have `tau_p`) and find better names for `a` and `m` (or are these letters used in some equations?)? 😅

**Comment by samiransen23** [comment]
It was a simple clarifying question...if `dihedrals` is supposed to empty, fine.
`tau` refers to thermostat --> usual. `tau_p` refers to barostat --> also usual. If it confuses you, you can make `tau` --> `tau_t`.
We cannot find better names for `a` .
We don't use `m` right now, and yes, it can be called `f`, if and when we use it.
@Lun4m

**Comment by mortele** [comment]
> It was a simple clarifying question...if `dihedrals` is supposed to empty, fine. `tau` refers to thermostat --> usual. `tau_p` refers to barostat --> also usual. If it confuses you, you can make `tau` --> `tau_t`. We cannot find better names for `a` . We don't use `m` right now, and yes, it can be called `f`, if and when we use it. @Lun4m

I am a bit apprehensive about changing the naming of `tau` now, since it will break existing version 1.0.x configuration scripts. However, we could maybe just deprecate `tau` and make `tau_t` the new default with `tau` being handled as a  synonym (until something like v1.2.0).

Should be straightforward to log a deprecation warning whenever `tau` is used in the config, and tell the user they should rename the variable to `tau_t`.

**Comment by mortele** [comment]
> It was a simple clarifying question...if `dihedrals` is supposed to empty, fine. `tau` refers to thermostat --> usual. `tau_p` refers to barostat --> also usual. If it confuses you, you can make `tau` --> `tau_t`. We cannot find better names for `a` . We don't use `m` right now, and yes, it can be called `f`, if and when we use it. @Lun4m

Even if Sigbjørn and you call it `a` in the equations in the pressure papers, that does not mean we are barred from renaming it to something more sane in the code. It is a number density used to fix the equation of state—might as well call it what it is.
